soccer/gameplay/skills/capture.py

```diff
@@ -152,5 +132,37 @@ def role_requirements(self):
         reqs.require_kicking = True
         # try to be near the ball
         if main.ball().valid:
-            reqs.destination_shape = main.ball().pos
+            reqs.cost_func = lambda r: reqs.position_cost_multiplier * find_robot_intercept_point(
+                r).dist_to(r.pos)
+
         return reqs
+
+
+def find_robot_intercept_point(robot):
+    if robot is None:
+        return main.ball().pos
+    approach_vec = approach_vector(robot)
+    # sample every 5 cm in the -approach_vector direction from the ball
+    pos = None
+    for i in range(50):
+        dist = i * 0.05
+        pos = main.ball().pos + approach_vec * dist
+        # how long will it take the ball to get there
+        ball_time = evaluation.ball.rev_predict(main.ball().vel, dist)
+        robotDist = (pos - robot.pos).mag() * 0.6
+        bot_time = robocup.get_trapezoidal_time(robotDist, robotDist, 2.2, 1,
```

Comment:
The 2.2 constant seems pretty important (max speed?). We should pull this from motion constratints at the very least